react-navigation/react-navigation
# feat: Enhance User Interaction with Customizable hitSlop on DrawerToggleButton

## Motivation

The motivation behind this change is to enhance the user experience by making the drawer toggle button more accessible, especially on devices with smaller screens or for users with accessibility needs. By allowing customizable `hitSlop` values, developers have greater control over the touchable area, ensuring a more user-friendly navigation experience.

The `hitSlop` was hardcoded with platform-specific values, leading to a limited and non-customizable touchable area. Specifically, the hitSlop for iOS was undefined, which could result in a challenging interaction experience for users on iOS devices due to the small target area.

The `hitSlop` property is now exposed as a prop, allowing for customizable configuration. Users can specify their preferred hitSlop values, or it defaults to { top: 16, right: 16, bottom: 16, left: 16 } for non-iOS platforms.

**_Impact_**
This change is backward compatible and does not introduce breaking changes. It adds an additional layer of customization for developers, catering to a wider range of user interactions and accessibility requirements.

I believe this enhancement aligns well with the goals of react-navigation to provide a robust and user-friendly navigation experience. I look forward to your feedback and suggestions on this PR.



**Test plan**

- Integrate the example code provided into your React Native application.
- Replace YOUR_CUSTOM_VALUE with different hitSlop configurations to test various scenarios. Example: { top: 20, right: 20, bottom: 20, left: 20 }.
- Start your React Native application on either an iOS or Android simulator/device.
- Navigate to the screen where the DrawerToggleButton is implemented.
- Experiment with different hitSlop values to test the touchable area around the DrawerToggleButton, try tapping around the edges of the button to ensure the extended touchable area works as expected.

### Example code

```
import {
  createDrawerNavigator,
  DrawerToggleButton
} from '@react-navigation/drawer'

const AppDrawer = () => {
const Drawer = createDrawerNavigator<AppNavigatorParamList>()

return (<Drawer.Navigator
      screenOptions={{
        headerLeft: () => <DrawerToggleButton hitSlop={YOUR_CUSTOM_VALUE} />
      }}>
      <Drawer.Screen name={routes.APP_HOME} component={app.Home} />
    </Drawer.Navigator>)
}
```
